modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/java/com/liferay/dynamic/data/mapping/form/builder/internal/context/DDMFormBuilderContextFactoryHelper.java

```diff
@@ -262,12 +263,24 @@ protected DDMFormValues doCreateDDMFormFieldSettingContextDDMFormValues(
 
 			ddmFormFieldValue.setName(propertyName);
 
-			DDMForm ddmForm = ddmFormField.getDDMForm();
+			Object propertyValue = ddmFormFieldProperties.get(propertyName);
 
-			Value value = doCreateDDMFormFieldValue(
-				ddmFormFieldTypeSetting,
-				ddmFormFieldProperties.get(propertyName),
-				ddmForm.getAvailableLocales());
+			Value value = null;
```

Comment:
I see that's your fix for the bug I've noticed in the previous PR. Looks good, but can you put this logic inside the `doCreateDDMFormFieldValue(DDMFormField ddmFormFieldTypeSetting, Object propertyValue, Set<Locale> availableLocales)` method? There's already some conditions to specific settings over there.

Also, this solution is only being applied to Forms, which means, Data Engine still faces the same problem as before. Please check Web Content Structures to see it happening. In order to fix that one, you'll have to include the same logic in the following method: `DataLayoutTaglibUtil._createDDMFormFieldValue(Set<Locale> availableLocales, DDMFormField ddmFormFieldTypeSetting, Object propertyValue)`.